dashboard/test/models/script_test.rb

```diff
@@ -1067,16 +1067,44 @@ def populate_cache_and_disconnect_db
     assert_equal 42, unit.peer_reviews_to_complete
 
     course_unit = unit.plc_course_unit
+    unit_group = unit.plc_course_unit.plc_course.unit_group
     assert_equal 'PLC Test', course_unit.unit_name
     assert_equal 'PLC test fixture script', course_unit.unit_description
 
+    assert_equal 'plc_reviewer', unit_group.instructor_audience
+    assert_equal 'facilitator', unit_group.participant_audience
+    assert_equal 'teacher_led', unit_group.instruction_type
+    assert_equal 'beta', unit_group.published_state
+
     lm = unit.lessons.first.plc_learning_module
     assert_equal 'Sample Module', lm.name
     assert_equal 1, course_unit.plc_learning_modules.count
     assert_equal lm, course_unit.plc_learning_modules.first
     assert_equal Plc::LearningModule::CONTENT_MODULE, lm.module_type
   end
 
+  test 'updating plc unit updates its unit group' do
+    Script.stubs(:unit_json_directory).returns(self.class.fixture_path)
+    unit = Script.seed_from_json_file('test-plc')
+
+    unit_group = unit.plc_course_unit.plc_course.unit_group
+
+    assert_equal 'plc_reviewer', unit_group.instructor_audience
+    assert_equal 'facilitator', unit_group.participant_audience
+    assert_equal 'teacher_led', unit_group.instruction_type
+    assert_equal 'beta', unit_group.published_state
+
+    unit.update!(instructor_audience: 'universal_instructor', participant_audience: 'teacher', instruction_type: 'self_paced', published_state: 'stable')
+
+    unit.reload
+    unit_group = unit.plc_course_unit.plc_course.unit_group
+
+    assert_equal 'universal_instructor', unit_group.instructor_audience
+    assert_equal 'teacher', unit_group.participant_audience
+    assert_equal 'self_paced', unit_group.instruction_type
+    assert_equal 'stable', unit_group.published_state
+  end
```

Comment:
are there any cases we care about where the plc unit already has a unit group (from a previous invocation of generate_plc_objects) and we need to change published state, instruction type or audience? My reading of `get_instruction_type` is that it would look to unit group first, and so a change to the unit's instruction type wouldn't actually get picked up.

Unless we are confident this will never happen, my suggestion would be to add a test case covering these values changing, whatever the desired behavior is in that case (e.g. it could be to raise an exception if these values are not supposed to change).

Reply:
Thanks for pushing on this. We should be just directly checking `instruction_type` instead of going to `get_instruction_type` here because we only care about what the script has set for instruction_type.

Reply:
per offline conversation, this seems good in terms of unit tests. there are some cases which may make sense to either add safeguards for or else carefully document:
* DLP courses are usually created by cloning. by default cloned courses start in the `in_development` state. need to be clear about how to get these into `beta` (either by clearly documenting that an engineer should do this right after cloning, or by modifying the clone script)